driver-core/src/main/com/mongodb/event/ConnectionCheckOutFailedEvent.java

```diff
@@ -53,19 +55,40 @@ public enum Reason {
     }
 
     private final ServerId serverId;
+    private final long operationId;
+    private final long elapsedTimeNanos;
+
     private final Reason reason;
 
     /**
      * Construct an instance
      *
      * @param serverId the server id
+     * @param operationId the operation id
+     * @param elapsedTimeNanos the elapsed time between check out started and check out failed
      * @param reason the reason the connection check out failed
+     * @since 4.10
      */
-    public ConnectionCheckOutFailedEvent(final ServerId serverId, final Reason reason) {
+    public ConnectionCheckOutFailedEvent(final ServerId serverId, final long operationId, final long elapsedTimeNanos,
```

Comment:
Is my understanding correct that there is no driver-wide requirement to provide elapsed time in events, and this is a Java-driver-specific initiative?

Reply:
That's right.  I've jumped the gun a bit on https://jira.mongodb.org/browse/DRIVERS-2469.  There is some risk that the designer for that ticket will propose something incompatible with this.

Reply:
Removed elapsedTime from PR